### ISSUE TITLE
include straight line when route not found

### ISSUE THREAD

**Issue by porjo** [Project-OSRM/osrm-backend#1391]
I've been using OSRM API to calculate a road route from a series of coordinates. Where no road route exists between 2 coordinates, the whole route fails with `Cannot find route between points`. It would be nice to have the option to override that behaviour and substitute a direct path between any 2 points where a route cannot be calculated. This would also be included in the geometry as a straight line between those points.

**Comment by DennisOSRM** [comment]
This is a design choice. Whenever a route is not found, the client is able to draw a straight line on its own trivially. There's no need to add redundant (and non-existent) geometry.

**Comment by porjo** [comment]
I may not have been clear in my original post. I'm talking specifically about the case where you may send dozens of coordinates in one go. A single non-routable segment causes the entire route to fail! 

The alternative would be to send a separate route request for each pair of coordinates. Sending them in series would take excessively long to complete, especially over high-latency links and sending them in parallel could result in server load issues. Being able to send them in a single batch, and have the server handle that gracefully would be ideal.

**Comment by TheMarex** [comment]
@porjo Are you trying to match a GPS trace to a road network? If so, what you actually want is to use a map matching algorithm. The problem is quite a bit more complex than just skipping points. :wink:
We are currently working on an plugin for map matching. ETA on that feature is about one month.

**Comment by porjo** [comment]
> Are you trying to match a GPS trace to a road network?

No. The application is a very simplistic trip planner which assumes road travel (when possible), and air travel anywhere else (for which a straight line between points would be sufficient to indicate the path taken). 

I had thought this might be a simple change, but if not - that's fine, I'll just work around it.

**Comment by DennisOSRM** [comment]
actually, it's simple to show _some_ route. but not the one that a user may (or may not) expect. Note that in general there's always two ways to traverse a road. And you only know the actual travel direction of each via location once you know how you arrive at the end. So, yes, we could guess something but the entire route may flip once you change a location in between.
